### ISSUE TITLE
BUG: `ElasticMaker` shouldn't use submaker as default factory

### ISSUE THREAD

**Issue by chiang-yuan** [materialsproject/atomate2#845]
`ElasticMaker` shouldn't use submaker as default factory otherwise `chgnet` will be required to be installed as dependency 

https://github.com/materialsproject/atomate2/blob/313f8098d58e3afcebd89a840aa489050afe2477/src/atomate2/forcefields/flows/elastic.py#L62-L70

**Comment by JaGeo** [comment]
> So currently, this doesn't actually require CHGNet to be installed I don't think. I.e., you can import this and overwrite those makers to get a MACE or M3GNet elastic constant workflow. However, it could make sense to create several versions of the workflow, one for each universal force field?
> 
> One thing I realised looking at this is that we can remove the [`prev_calc_dir_argname` function](https://github.com/materialsproject/atomate2/blob/aee4be89b8a16a0f2a79e0c9aa22e3859826b4b9/src/atomate2/common/flows/elastic.py#L159C9-L159C30) from the `BaseElasticMaker` and subclasses since this has now been standardised across all codes.

@utf also in the Abinit workflows?

**Comment by chiang-yuan** [comment]
Yes @utf you are definitely right - one could always replace default factory with the relaxmaker they like to use. However, I think this design is a bit counterintuitive and one might just want to set the forcefield calculator name and the code switch the input generator accordingly like what we did for `MDMaker`

Not a serious issue so maybe I shouldn't call it bug at the first place (I think it is the template..)

**Comment by esoteric-ephemera** [comment]
Confirming this doesn't require `CHGNet` to be installed, as the import only happens once an `ase_calculator` is called (when the job actually runs)

We don't have a good solution for general initialization for forcefield classes, since there's no "default" / testbed forcefield. Might be worth doing something like having a "generic universal potential" (ZBL, UFF?) as the default

Also @JaGeo  is correct: the abinit workflows don't use conventional `prev_dir` syntax in their `make` functions. The  `prev_calc_dir_argname` might have to stay for now

Closing for now, this could be something we look at in the future
